src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala

```diff
@@ -35,11 +35,14 @@ object XSDToSchema {
    * Reads a schema from an XSD file.
    *
    * @param xsdFile XSD file
+   * @param baseURI base URI for relative imports
    * @return Spark-compatible schema
    */
   @Experimental
-  def read(xsdFile: File): StructType = {
-    val xmlSchema = new XmlSchemaCollection().read(
+  def fromFile(xsdFile: File, baseURI: Option[String] = None): StructType = {
```

Comment:
Though it's "Experimental" I'd prefer not changing the method names just for its own sake

Reply:
Scala compiler does not allow for having default arguments on multiple overloaded methods

Reply:
Ah hm OK. We can just add a new method too, which even helps preserve binary compatibility. I'm neutral on it I suppose.

Reply:
What does binary compatibility has to do with this ? 
On the other hand this is still marked `Experimental` and not even released. If you are looking to change it, now might be the best time :)

Reply:
The one-arg File method would go away entirely so apps that link against that would fail. Source compatibility is fine though. 
But yes oh I forgot this isn't even released! So I agree, renaming is entirely fine.